runtime/src/main/java/io/micronaut/logging/LogbackLogLevelConfigurer.java

```diff
@@ -0,0 +1,92 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.logging;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.annotation.Context;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.env.Environment;
+import io.micronaut.context.event.ApplicationEventListener;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.naming.conventions.StringConvention;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.runtime.context.scope.refresh.RefreshEvent;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+
+/**
+ * Logback log level configurer.
+ *
+ * @author Denis Stepanov
+ * @since 1.3.0
+ */
+@BootstrapContextCompatible
+@Singleton
+@Context
+@Requires(classes = ch.qos.logback.classic.LoggerContext.class)
+@Requires(beans = Environment.class)
+@Requires(property = LogbackLogLevelConfigurer.LOGGER_LEVELS_PROPERTY_PREFIX)
+@Internal
+final class LogbackLogLevelConfigurer implements ApplicationEventListener<RefreshEvent> {
+
+    static final String LOGGER_LEVELS_PROPERTY_PREFIX = "logger.levels";
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(LogbackLogLevelConfigurer.class);
+
+    private final Environment environment;
+
+    /**
+     * Sets log level according to properties.
+     *
+     * @param environment The  environment
+     */
+    public LogbackLogLevelConfigurer(Environment environment) {
+        this.environment = environment;
+        configureLogLevels(environment);
+    }
+
+    /**
+     * Sets log level according to properties on refresh.
+     *
+     * @param event refresh event
+     */
+    @Override
+    public void onApplicationEvent(RefreshEvent event) {
+        configureLogLevels(environment);
+    }
+
+    private void configureLogLevels(Environment environment) {
+        environment.getProperties(LOGGER_LEVELS_PROPERTY_PREFIX, StringConvention.RAW).forEach((loggerPrefix, levelString) -> {
+            Logger logger = (Logger) LoggerFactory.getLogger(loggerPrefix);
+            Level newLevel = toLevel(levelString.toString());
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Setting log level {} to {}", loggerPrefix, newLevel);
+            }
+            logger.setLevel(newLevel);
+        });
+    }
+
+    private static Level toLevel(String logLevel) {
+        if (StringUtils.isEmpty(logLevel)) {
+            return null;
+        } else {
+            return Level.valueOf(logLevel);
```

Comment:
Think we need to catch the exception of `logLevel` is invalid and either log a warning or rethrow as `ConfigurationException`

Reply:
Added